src/parse.c

```diff
@@ -240,6 +240,8 @@ netplan_netdef_new(const char* id, NetplanDefType type, NetplanBackend backend)
     /* OpenVSwitch defaults */
     initialize_ovs_settings(&cur_netdef->ovs_settings);
 
+    if (!netdefs)
+        netdefs = g_hash_table_new(g_str_hash, g_str_equal);
```

Comment:
Wondering if since we now allocate the `netdefs` table in `netplan_netdef_new()`, maybe we could get rid of it from `netplan_parse_yaml()`? Since I don't think we put anything into `netdefs` before at least once calling `netplan_netdef_new()`.

Reply:
Yes, I was wondering about this as well. Although, we do not put anything into `netdefs` before, there are a few places where we expect a pre-deployed hashmap for reading before. I've now figured out all of those spots and handled them accordingly.